test/test_cpp20/hstring.cpp

```diff
@@ -0,0 +1,13 @@
+#include "pch.h"
+
+TEST_CASE("hstring")
+{
+	winrt::hstring text = L"C++/WinRT rocks!";
+	std::wstring_view textView = text;
+	REQUIRE(!text.starts_with(L"C++/CX"));
+	REQUIRE(!textView.starts_with(L"C++/CX"));
+	REQUIRE(text.starts_with(L"C++/WinRT") == textView.starts_with(L"C++/WinRT"));
+	REQUIRE(text.ends_with("rocks!"));
```

Comment:
Please rerun the tests locally to make sure it builds. Here you've missing a `L` and also below.

The way I do this is to build an x64 Release build of cppwinrt in Visual Studio. Then at the command prompt you can run `build_projection.cmd`. Then back in Visual Studio you can switch to x64 Debug and then build the entire solution. That should smoke out any build issues like this. Then finally you can actually run all of the tests at the command line with `run_tests.cmd` that will highly issues caught at run time.

Reply:
I'm getting a few errors in some other files: 14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(103,74): error C2641: cannot deduce template arguments for 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(103,74): error C2783: 'winrt::weak_ref<T> winrt::weak_ref(const conditional<std::is_base_of_v<winrt::Windows::Foundation::IUnknown,T>,T,winrt::com_ptr<T>>::type &)': could not deduce template argument for 'T'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4109): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(103,74): error C2783: 'winrt::weak_ref<T> winrt::weak_ref(std::nullptr_t) noexcept': could not deduce template argument for 'T'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4107): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(103,88): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::IStringable'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(103,88): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::IStringable'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(103,88): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::IStringable'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(103,88): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::IStringable'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(103,88): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::IStringable'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(103,88): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::IStringable'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(103,88): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::IStringable'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(103,88): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::IStringable'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(103): error C2062: type 'unknown-type' unexpected
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104,66): error C2641: cannot deduce template arguments for 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104,66): error C2783: 'winrt::weak_ref<T> winrt::weak_ref(const conditional<std::is_base_of_v<winrt::Windows::Foundation::IUnknown,T>,T,winrt::com_ptr<T>>::type &)': could not deduce template argument for 'T'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4109): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104,66): error C2783: 'winrt::weak_ref<T> winrt::weak_ref(std::nullptr_t) noexcept': could not deduce template argument for 'T'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4107): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104,86): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::Uri'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104,86): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::Uri'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104,86): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::Uri'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104,86): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::Uri'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104,86): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::Uri'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104,86): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::Uri'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104,86): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::Uri'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104,86): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::Uri'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104,86): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::Uri'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104,86): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::Uri'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104,86): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::Uri'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104,86): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::Uri'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104,86): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::Uri'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104,86): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::Uri'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104,86): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::Uri'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104,86): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::Uri'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104,86): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::Uri'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104,86): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::Uri'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104,86): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::Uri'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104,86): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::Windows::Foundation::Uri'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(104): error C2062: type 'unknown-type' unexpected
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(105,73): error C2641: cannot deduce template arguments for 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(105,73): error C2783: 'winrt::weak_ref<T> winrt::weak_ref(const conditional<std::is_base_of_v<winrt::Windows::Foundation::IUnknown,T>,T,winrt::com_ptr<T>>::type &)': could not deduce template argument for 'T'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4109): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(105,73): error C2783: 'winrt::weak_ref<T> winrt::weak_ref(std::nullptr_t) noexcept': could not deduce template argument for 'T'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4107): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(105,95): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::com_ptr<IPersist>'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(105,95): error C2784: 'winrt::weak_ref<T> winrt::weak_ref(winrt::weak_ref<T>)': could not deduce template argument for 'winrt::weak_ref<T>' from 'winrt::com_ptr<IPersist>'
14>C:\Users\jaiga\Desktop\source\cppwinrt\_build\x64\Debug\winrt\base.h(4106): message : see declaration of 'winrt::weak_ref'
14>C:\Users\jaiga\Desktop\source\cppwinrt\test\old_tests\UnitTests\weak.cpp(105): error C2062: type 'unknown-type' unexpected

Reply:
This indicates that the build didn't find the headers generated by `build_projection` for the target configuration and instead fell back to using the old headers in the Windows SDK.

Reply:
I have VS 16.10.2. Isn't that the latest?

Reply:
Yes, that sounds right.

Reply:
All I can suggest is you double check the steps here: https://github.com/microsoft/cppwinrt#building-cwinrt

Reply:
It only seems to occur in the old test project, other projects were built successfully.

Reply:
Works for me. 😉 

```
**********************************************************************
** Visual Studio 2019 Developer Command Prompt v16.10.2
** Copyright (c) 2021 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64'

C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise>cd c:\git\cppwinrt

c:\git\cppwinrt>msbuild /m /p:Configuration=Debug,Platform=x64 cppwinrt.sln /t:test\test_cpp20
Microsoft (R) Build Engine version 16.10.2+857e5a733 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 6/22/2021 7:07:14 AM.
     1>Project "c:\git\cppwinrt\cppwinrt.sln" on node 1 (test\test_cpp20 target(s)).
     1>ValidateSolutionConfiguration:
         Building solution configuration "Debug|x64".
     1>Project "c:\git\cppwinrt\cppwinrt.sln" (1) is building "c:\git\cppwinrt\test\test_cpp20\test_cpp20.vcxproj.metaproj" (2) on node 1 (default targets).
     2>Project "c:\git\cppwinrt\test\test_cpp20\test_cpp20.vcxproj.metaproj" (2) is building "c:\git\cppwinrt\cppwinrt\cppwinrt.vcxproj.metaproj" (3) on node 1 (default targets).
     3>Project "c:\git\cppwinrt\cppwinrt\cppwinrt.vcxproj.metaproj" (3) is building "c:\git\cppwinrt\prebuild\prebuild.vcxproj" (4) on node 1 (default targets).
     4>InitializeBuildStatus:
         Creating "c:\git\cppwinrt\_build\x64\Debug\temp\prebuild\prebuild.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
       ClCompile:
         All outputs are up-to-date.
         All outputs are up-to-date.
       Link:
         All outputs are up-to-date.
         prebuild.vcxproj -> c:\git\cppwinrt\_build\x64\Debug\prebuild.exe
       FinalizeBuildStatus:
         Deleting file "c:\git\cppwinrt\_build\x64\Debug\temp\prebuild\prebuild.tlog\unsuccessfulbuild".
         Touching "c:\git\cppwinrt\_build\x64\Debug\temp\prebuild\prebuild.tlog\prebuild.lastbuildstate".
     4>Done Building Project "c:\git\cppwinrt\prebuild\prebuild.vcxproj" (default targets).
     3>Project "c:\git\cppwinrt\cppwinrt\cppwinrt.vcxproj.metaproj" (3) is building "c:\git\cppwinrt\cppwinrt\cppwinrt.vcxproj" (5) on node 1 (default targets).
     5>InitializeBuildStatus:
         Creating "c:\git\cppwinrt\_build\x64\Debug\temp\cppwinrt\cppwinrt.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
       PreBuildEvent:
         c:\git\cppwinrt\_build\x64\Debug\prebuild.exe ..\strings c:\git\cppwinrt\_build\x64\Debug\
         :VCEnd
       ClCompile:
         All outputs are up-to-date.
         All outputs are up-to-date.
         All outputs are up-to-date.
       ResourceCompile:
         All outputs are up-to-date.
       ManifestResourceCompile:
         All outputs are up-to-date.
       Link:
         All outputs are up-to-date.
         cppwinrt.vcxproj -> c:\git\cppwinrt\_build\x64\Debug\cppwinrt.exe
       FinalizeBuildStatus:
         Deleting file "c:\git\cppwinrt\_build\x64\Debug\temp\cppwinrt\cppwinrt.tlog\unsuccessfulbuild".
         Touching "c:\git\cppwinrt\_build\x64\Debug\temp\cppwinrt\cppwinrt.tlog\cppwinrt.lastbuildstate".
     5>Done Building Project "c:\git\cppwinrt\cppwinrt\cppwinrt.vcxproj" (default targets).
     3>Done Building Project "c:\git\cppwinrt\cppwinrt\cppwinrt.vcxproj.metaproj" (default targets).
     2>Project "c:\git\cppwinrt\test\test_cpp20\test_cpp20.vcxproj.metaproj" (2) is building "c:\git\cppwinrt\test\test_cpp20\test_cpp20.vcxproj" (6) on node 1 (default targets).
     6>InitializeBuildStatus:
         Creating "c:\git\cppwinrt\_build\x64\Debug\temp\test_cpp20\test_cpp20.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
       ClCompile:
         All outputs are up-to-date.
         All outputs are up-to-date.
         All outputs are up-to-date.
       Link:
         All outputs are up-to-date.
         test_cpp20.vcxproj -> c:\git\cppwinrt\_build\x64\Debug\test_cpp20.exe
       FinalizeBuildStatus:
         Deleting file "c:\git\cppwinrt\_build\x64\Debug\temp\test_cpp20\test_cpp20.tlog\unsuccessfulbuild".
         Touching "c:\git\cppwinrt\_build\x64\Debug\temp\test_cpp20\test_cpp20.tlog\test_cpp20.lastbuildstate".
     6>Done Building Project "c:\git\cppwinrt\test\test_cpp20\test_cpp20.vcxproj" (default targets).
     2>Done Building Project "c:\git\cppwinrt\test\test_cpp20\test_cpp20.vcxproj.metaproj" (default targets).
     1>Done Building Project "c:\git\cppwinrt\cppwinrt.sln" (test\test_cpp20 target(s)).

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.86
```

Reply:
test_cpp20 works for me also but test_old doesn't. Can I ignore it?

Reply:
'_build\x64\Debug\test.exe' is not recognized as an internal or external command,
operable program or batch file.
===============================================================================
All tests passed (10 assertions in 2 test cases)

===============================================================================
All tests passed (1011 assertions in 47 test cases)

===============================================================================
All tests passed (8 assertions in 2 test cases)

===============================================================================
All tests passed (11 assertions in 2 test cases)

'_build\x64\Debug\test_old.exe' is not recognized as an internal or external command,
operable program or batch file.
===============================================================================
All tests passed (3 assertions in 1 test case)

===============================================================================
All tests passed (4 assertions in 1 test case)

All tests were passed except old_test and test which weren't built but I'll ignore that for now.